test/functional/ui/output_spec.lua

```diff
@@ -39,4 +40,11 @@ describe("shell command :!", function()
       {3:-- TERMINAL --}                                    |
     ]])
   end)
+
+  it("throttles shell-command output greater than ~20KB", function()
+    child_session.feed_data(
+      ":!for i in $(seq 2 3000); do echo XXXXXXXXXX; done\n")
+    -- If a line with only a dot "." appears, then throttling was triggered.
+    screen:expect("\n.", nil, nil, nil, true)
```

Comment:
Passing `any=true` to `screen:expect` checks that the string appears "somewhere" in the screen state. The expected string `"\n."` means that throttling was triggered for the shell output.